src/NumericalAlgorithms/Interpolation/CubicSpline.cpp

```diff
@@ -0,0 +1,62 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/Interpolation/CubicSpline.hpp"
+
+#include <cstddef>
+#include <gsl/gsl_spline.h>
+#include <pup.h>
+
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace intrp {
+CubicSpline::CubicSpline(std::vector<double> x_values,
+                         std::vector<double> y_values) noexcept
+    : x_values_(std::move(x_values)), y_values_(std::move(y_values)) {
+  ASSERT(x_values_.size() == y_values_.size(),
+         "The x-value and y-value vectors must be of the same length, but "
+         "received x-value of size: "
+             << x_values_.size()
+             << " and y-value of size: " << y_values_.size());
+  initialize_interpolant();
+}
+
+void CubicSpline::gsl_interp_accel_deleter::operator()(
+    gsl_interp_accel* acc) noexcept {
```

Comment:
can the pointer and the call operator not be `const`?

Reply:
the call operator can be `const, but the pointer can't be:

```cpp
/usr/include/gsl/gsl_spline.h:97:1: note: candidate function not viable: 1st argument ('const gsl_spline *') would lose const qualifier
gsl_spline_free(gsl_spline * spline);
```